kuma/javascript/src/bcd-signal.jsx

```diff
@@ -0,0 +1,809 @@
+// @flow
+import { gettext, interpolate } from './l10n.js';
+import { getCookie } from './utils.js';
+
+let bcSignalsRendered = false;
+
+export function activateBCDSignals(slug: string, locale: string) {
+    if (bcSignalsRendered) {
+        return;
+    }
```

Comment:
Why not just do:
```
export function activateBCDSignals(slug: string, locale: string) {
    if (bcSignalsRendered) {
        return;
    }
    bcSignalsRendered = true;
```
Perhaps rename that variable to `bcSignalsInvoked`. 

I can't yet see if that's realistic but what if this function gets called twice within a short span of time, async. Since the setting of `bcSignalsRendered = true;` happens 800 lines away from here, that second invocation might come sooner.

Reply:
You have a point. As a general rule I prefer setting the boolean after the component is invoked and rendered (who knows what happens in the process), but in this case your concern is valid and probably moving it is a good idea.